ql/src/java/org/apache/hadoop/hive/ql/exec/tez/Utils.java

```diff
@@ -96,14 +98,15 @@ public String[] getLocations(InputSplit split) throws IOException {
   }
 
   @VisibleForTesting
-  static SplitLocationProvider getCustomSplitLocationProvider(LlapRegistryService serviceRegistry, Logger LOG) throws
-      IOException {
+  static SplitLocationProvider getCustomSplitLocationProvider(LlapRegistryService serviceRegistry,
+      int clusterReadyTimeoutMs, Logger LOG) throws IOException {
     LOG.info("Using LLAP instance " + serviceRegistry.getApplicationId());
 
     Collection<LlapServiceInstance> serviceInstances =
-        serviceRegistry.getInstances().getAllInstancesOrdered(true);
+        serviceRegistry.getInstances(clusterReadyTimeoutMs).getAllInstancesOrdered(true);
```

Comment:
okay, I've just realized this is not what we want
clusterReadyTimeoutMs is for zookeeper readiness, but we want to wait x ms until at least 1 LLAP instance is up, which needs custom logic here (some polling)